core-bundle/src/Util/UrlUtil.php

```diff
@@ -12,11 +12,35 @@
 
 namespace Contao\CoreBundle\Util;
 
+use Contao\CoreBundle\InsertTag\InsertTagParser;
 use Nyholm\Psr7\Uri;
 use Symfony\Component\Filesystem\Path;
+use Symfony\Component\HttpFoundation\RequestStack;
 
 class UrlUtil
 {
+    public function __construct(
+        private readonly InsertTagParser $insertTagParser,
+        private readonly RequestStack $requestStack,
+    ) {
+    }
+
+    public function parseContaoUrl(string $url): string
+    {
+        $url = $this->insertTagParser->replaceInline($url);
+
+        // Ensure absolute links
+        if (!preg_match('#^https?://#', $url)) {
+            if (!$request = $this->requestStack->getCurrentRequest()) {
+                throw new \RuntimeException('The request stack did not contain a request');
+            }
+
+            $url = static::makeAbsolute($url, $request->getUri());
```

Comment:
This should be optional and controllable via parameter. It was only there because the code you copied is for the canonical URL which has to be absolute. Many other usages propbably don’t need an absolute URL.

Reply:
I am pretty sure we meant path absolute URLs.